metals/src/main/scala/scala/meta/internal/metals/Compilations.scala

```diff
@@ -222,7 +222,7 @@ final class Compilations(
         CancelableFuture.sequence(futures).map(_.flatten.toMap)
     }
   }
-  private def compile(
+  def compile(
```

Comment:
I think you can just add a `compileAll` method based on `recompileAll` without the cleaning and use that. You will need only to add Compilations to the DebugProvider.